File: src/main/java/io/github/sst/remake/event/impl/game/render/Render2DEvent.java
package io.github.sst.remake.event.impl.game.render;

import io.github.sst.remake.event.Event;
import lombok.RequiredArgsConstructor;

/**
 * @apiNote DON'T USE THIS FOR REGULAR UI RENDERING, INSTEAD USE
 * @see io.github.sst.remake.event.impl.client.RenderClient2DEvent
 */
@RequiredArgsConstructor
public class Render2DEvent extends Event {
    public final float tickDelta;
    public final long startTime;
    public final boolean tick;
}

File: src/main/kotlin/info/opensigma/event/Render2DEvent.kt
package info.opensigma.event;


public class Render2DEvent {
    public final float tickDelta;
    public final long startTime;

    public Render2DEvent() {
        this.startTime = 0L;
        this.tickDelta = 0.0F;
    }

    public Render2DEvent(float tickDelta, long startTime) {
        this.startTime = startTime;
        this.tickDelta = tickDelta;
    }
}